### PR TITLE
More optimizations

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java [hunk 1]
```diff
@@ -292,40 +307,30 @@ private void writeToJson(String name, String value, int len) throws IOException
...
-    private <T extends JsonSerializable> String createJsonFor(T value) throws IOException {
-        StringWriter stringWriter = new StringWriter();
-        JsonTelemetryDataSerializer temp = new JsonTelemetryDataSerializer(new BufferedWriter(stringWriter));
-
-        value.serialize(temp);
-        temp.close();
-        String jsonStringToAppend = stringWriter.toString();
-        if (Strings.isNullOrEmpty(jsonStringToAppend) || JSON_EMPTY_OBJECT.equals(jsonStringToAppend)) {
-            return "";
+    private <T> void writeObject(JsonSerializable value) throws IOException {
```
Comment (littleaj) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java]: Looks like we don't need the type variable anymore

File: agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/IncomingSpanImpl.java [hunk 2]
```diff
@@ -227,7 +217,7 @@ private void finishBuildingWebTelemetry(long endTimeMillis) {
             requestTelemetry.setSuccess(responseCode < 400);
         }
 
-        processCookies(detail);
+        // processCookies(detail);
```
Comment (littleaj) [inline agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/IncomingSpanImpl.java]: should this be removed?

File: core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java [hunk 3]
```diff
@@ -31,6 +31,7 @@
 
 import java.io.IOException;
 import java.io.StringWriter;
+import java.lang.ref.WeakReference;
```
Comment (littleaj) [inline core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java]: Unused

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java [hunk 4]
```diff
@@ -45,7 +45,9 @@
     /**
      * Backing field for property Name.
      */
-    private String name;
+    private String namePart1;
+    private String namePart2;
+    private String namePart3;
```
Comment (littleaj) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java]: It took a minute to find what part 1, 2 and 3 are. Let's name them something like prefix, normalizedIkey, telemetryType or document in javadoc.

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java [hunk 5]
```diff
@@ -224,12 +230,7 @@ public static String normalizeInstrumentationKey(String instrumentationKey){
     }
 
     public static String getTelemetryName(String normalizedInstrumentationKey, String envelopType){
-        return String.format(
-                "%s%s%s",
-                TELEMETRY_NAME_PREFIX,
-                normalizedInstrumentationKey,
-                envelopType
-                );
+        return TELEMETRY_NAME_PREFIX + normalizedInstrumentationKey + envelopType;
```
Comment (littleaj) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: 👍

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java (hunk 1 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java]: 👍

File: core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java (hunk 3 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java]: 👍

**Comment by trask** [comment]
Sorry, I meant to submit this as a draft PR. Some of this work is on top of #1111. I addressed your comments above in #1111, and will come back to this PR once that is merged. Thanks!

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java (hunk 4 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java]: 👍

File: agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/IncomingSpanImpl.java (hunk 2 above)
Comment (trask) [reply agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/IncomingSpanImpl.java]: 👍

**Comment by trask** [comment]
Closing, replaced by #1122 and #1123

**Pull request closed without merging**